pkg/networkaware/networkoverhead/networkoverhead_test.go

```diff
@@ -0,0 +1,1546 @@
+			dependenciesNum: 10,
+			agName:          "onlineboutique",
+			regionNames:     regionNames,
+			zoneNames:       zoneNames,
+			appGroup:        onlineBoutiqueAppGroup,
+			networkTopology: networkTopology,
+			pod:             makePod("P1", "P1-deployment", 0, "onlineboutique", nil, nil),
+			pods:            pods,
+		},
+		{
+			name:            "AppGroup: onlineboutique, 10 pods allocated, 10000 nodes, 1 pod to allocate",
+			nodesNum:        10000,
+			dependenciesNum: 10,
+			agName:          "onlineboutique",
+			regionNames:     regionNames,
+			zoneNames:       zoneNames,
+			appGroup:        onlineBoutiqueAppGroup,
+			networkTopology: networkTopology,
+			pod:             makePod("P1", "P1-deployment", 0, "onlineboutique", nil, nil),
+			pods:            pods,
+		},
+	}
+
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			// init listers
+			agClient := agfake.NewSimpleClientset()
+			ntClient := ntfake.NewSimpleClientset()
+
+			fakeAgInformer := aginformers.NewSharedInformerFactory(agClient, 0).Appgroup().V1alpha1().AppGroups()
+			fakeNTInformer := ntinformers.NewSharedInformerFactory(ntClient, 0).Networktopology().V1alpha1().NetworkTopologies()
+
+			// init nodes
+			nodes := getNodes(tt.nodesNum, tt.regionNames, tt.zoneNames)
+
+			// Create dependencies
+			tt.appGroup.Status.RunningWorkloads = tt.dependenciesNum
+
+			// add CRDs
+			agLister := fakeAgInformer.Lister()
+			ntLister := fakeNTInformer.Lister()
+
+			_ = fakeAgInformer.Informer().GetStore().Add(tt.appGroup)
+			_ = fakeNTInformer.Informer().GetStore().Add(tt.networkTopology)
+
+			// create plugin
+			ctx := context.Background()
+			cs := testClientSet.NewSimpleClientset()
+
+			informerFactory := informers.NewSharedInformerFactory(cs, 0)
+
+			snapshot := newTestSharedLister(nil, nodes)
+
+			podInformer := informerFactory.Core().V1().Pods()
+			podLister := podInformer.Lister()
+			informerFactory.Start(ctx.Done())
+
+			for _, p := range tt.pods {
+				_, err := cs.CoreV1().Pods("default").Create(ctx, p, metav1.CreateOptions{})
+				if err != nil {
+					b.Fatalf("Failed to create Workload %q: %v", p.Name, err)
+				}
+				//b.Logf("Workload %v created  \n", p.Name)
+			}
+
+			registeredPlugins := []st.RegisterPluginFunc{
+				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
+				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
+			}
+
+			fh, _ := st.NewFramework(registeredPlugins, "default-scheduler", runtime.WithClientSet(cs),
+				runtime.WithInformerFactory(informerFactory), runtime.WithSnapshotSharedLister(snapshot))
+
+			pl := &NetworkOverhead{
+				handle:      fh,
+				agLister:    agLister,
+				podLister:   podLister,
+				ntLister:    ntLister,
+				namespaces:  []string{"default"},
+				weightsName: "UserDefined",
+				ntName:      "nt-test",
+			}
+
+			state := framework.NewCycleState()
+
+			// Without sleep sometimes the pods are not created in time
+			time.Sleep(1 * time.Second)
```

Comment:
ditto. we should avoid using time.Sleep() for unexpected flakiness.

Reply:
What do you think of the following change? Is there a cleaner way? 

```go
// Wait for the pods to be scheduled.
if err := wait.Poll(1*time.Second, 20*time.Second, func() (bool, error) {
return true, nil
}); err != nil {
b.Errorf("pods not scheduled yet: %v ", err)
}
```